### PR TITLE
Cleaning up getting started, install, architecture and community chapters

### PR DESCRIPTION
**Pull request by sandeep-krishnamurthy**
* Revamped installation guide fully. Complete clean up of language and organization of content.
* Re-organized getting started page.
* Re-organized system design and architecture pages
* Created a new community page.
* Added "Recommended Steps" sections at end of pages. This helps users to logically navigate section after sections.

### CODE REVIEW

**Comment by sandeep-krishnamurthy** [comment]
@piiswrong : Please do let me know if you need more details for merging these changes.

File: docs/architecture/index.md [hunk 1]
```diff
@@ -1,86 +1,32 @@
...
-* [Squeeze the Memory Consumption of Deep Learning](note_memory.md)
-* [Efficient Data Loading Module for Deep Learning](note_data_loading.md)
-* [Survey of RNN Interface](rnn_interface.md)
 
-The next parts will be specific to MXNet
+&nbsp;
 
-## Overview of the Design
+Refer below sections for more details
+* [Programming Models for Deep Learning](http://mxnet.io/architecture/program_model.html)
```
Comment (piiswrong) [inline docs/architecture/index.md]: can we have a 1 to 5 sentence intro for all the links in the next iteration? Similar to TF

File: docs/architecture/overview.md [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+    mshadow::Tensor<xpu, 2, DType> src = in_data0.data.get<xpu, 2, DType>(s);
+    mshadow::Tensor<xpu, 2, DType> ograd = out_grad.data.get<xpu, 2, DType>(s);
+    mshadow::Tensor<xpu, 2, DType> igrad = in_grad->get<xpu, 2, DType>(s);
+    ASSIGN_DISPATCH(igrad, req,
+                    ograd * F<mshadow_op::smooth_l1_gradient>(src, ScalarExp<DType>(sigma2)));
+  });
+}
+```
+
+### Register SimpleOp to MXNet
```
Comment (piiswrong) [inline docs/architecture/overview.md]: mark this as deprecated

**Comment by piiswrong** [comment]
@sandeep-krishnamurthy title and indentation on this page is in consistent: http://mxnet.io/get_started/setup.html

File: docs/architecture/index.md (hunk 1 above)
Comment (sandeep-krishnamurthy) [inline docs/architecture/index.md]: Yes this is important. I will take up in next iteration. 
With this change, I am bringing all related contents together and adding "recommended next steps" to guide users logically next topics.

File: docs/architecture/overview.md (hunk 2 above)
Comment (sandeep-krishnamurthy) [inline docs/architecture/overview.md]: Yes will do.

**Comment by sandeep-krishnamurthy** [comment]
@piiswrong This is mainly CSS issue which we need to take up. With this change we now have very good TOC side bar on left to help user navigate easily.

We need following 2 things to fix issues all across the documents:
1. Bullet list should have indentation.
2. We need to choose "font" size and style which is better than the current one.

Is there any specific indentation issue apart from above?

**Comment by sandeep-krishnamurthy** [comment]
Added more details in this issue https://github.com/dmlc/mxnet/issues/3531

**Pull request merged**